src/test/run-pass/invalid_const_promotion.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// ignore-wasm
+
+#![feature(const_fn)]
+#![allow(const_err)]
+
+use std::env;
+use std::process::{Command, Stdio};
+
+const fn bar() -> usize { 0 - 1 }
+
+fn foo() {
+    let _: &'static _ = &bar();
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    if args.len() > 1 && args[1] == "test" {
+        foo();
+        return;
+    }
+
+    let mut p = Command::new(&args[0])
+        .stdout(Stdio::piped())
+        .stdin(Stdio::piped())
+        .arg("test").output().unwrap();
+    assert!(!p.status.success());
```

Comment:
This test would also pass if the program SEGFAULTs (like some variants of this did) instead of SIGILLs... but I am not sure how to test for this properly, so whatever.^^

Reply:
https://github.com/rust-lang/rust/blob/a5c2d0fffaaf0b764c01bc4066e51ffd475ceae9/src/test/run-pass/segfault-no-out-of-stack.rs#L26-L40

Reply:
So...? That's how this test could be improved, is what you are saying?

Reply:
I mean we already have copy-pastable code that does the exact checking necessary here. We should avoid writing non-trivial tests by hand when there's already a good example.